### DISCUSSION TITLE
Base

### DISCUSSION THREAD

**Discussion by kinlane** [postman-open-technologies/lifecycle#27] [General]
This is a conversation thread for the base blueprint. It is meant to help iterate and evolve this blueprint along with pull requests or issues. Here are the two relevant links for this blueprint:

- [Detail Page](https://postman-open-technologies.github.io/lifecycle/blueprints/base/)
- [YAML Source](https://github.com/postman-open-technologies/lifecycle/blob/main/_blueprints/base.md)

Leave all questions, comments, and feedback here, and go ahead and submit pull requests to make any changes.

**Comment by DavidBiesack** [comment]
- I'd like to see taxonomy and terminology addressed. i.e. for search to work, we need to use the same terms, or have some way to map terms.
- Usually "partner" is included in the spectrum of public/partner/private APIs (Discover, Publish)
- Interesting that Security addresses RBAC around the lifecycle, but not designing security into the API itself (i.e. address security early in the lifecycle). Could fit w/i "Goals" of Design (we use documented API Design Principles to guide API design)
- Deprecation applies to the building blocks as well as entire APIs - operations, schemas, etc. (At Apiture, we put semantic versions on schemas as well as APIs, but not yet on operations or other elements, as we don't reuse - yet -  those as building blocks as we do with schemas)

**Comment by DavidBiesack** [comment]
copy/edit:
- Is there a verb missing in the "Private API networks all organizations, groups, and teams" ? I could not parse.
- "iterating upon API" s/b "iterating upon APIs"
- "parts nd pieces" s/b "parts nd pieces"
- "so they and collaborate" s/b "so they can collaborate"
- "learn about what an API does" s/b "learn what an API does"
- It's unclear if "building blocks", "elements of APIs" are the same or different concepts
- "usefulness to automation" feels awkward to me. "scriptability"?
- "An APIs usefulness to automation" s/b "An API's ...."
- "allowing for in-application, email, or type of message" s/b "allowing for in-application, email, or other type of message"
- "So too, do APIs." s/b "So too do APIs."
- "a formal process to the deprecation of APIs" is awkward; do you mean "a formal process to deprecate APIs"  or "a formal process for the deprecation of APIs" ?
- the "YAML" icon links to .md not .yaml but that .md page appears to be YAML
- 
-

**Comment by kinlane** [comment]
This is such great feedback @DavidBiesack -- very much appreciated. Adding as task to work through these points. Would love to chat sometime about your high level view of the lifecycle from your position.
